report_aeroo/report_parser.py

```diff
@@ -137,104 +140,113 @@ def get_selection_item(obj, field, value=None):
             return
         icp = self.env.get('ir.config_parameter')
         icpgp = icp.get_param
         docs_host = icpgp('aeroo.docs_host') or 'localhost'
         docs_port = icpgp('aeroo.docs_port') or '8989'
-        docs_auth_type = icpgp('aeroo.docs_auth_type') or False
+        # docs_auth_type = icpgp('aeroo.docs_auth_type') or False
         docs_username = icpgp('aeroo.docs_username') or 'anonymous'
         docs_password = icpgp('aeroo.docs_password') or 'anonymous'
-        docs_client = DOCSConnection(docs_host, docs_port,
-                        username=docs_username, password=docs_password)
+        docs_client = DOCSConnection(
+            docs_host, docs_port, username=docs_username,
+            password=docs_password)
         self.docs_client = docs_client
-    
+
     def _generate_doc(self, data, report):
         docs = self.docs_client
         token = docs.upload(data)
-        if report.out_format.code=='oo-dbf':
-            data = docs.convert(identifier=token) #TODO What format?
+        if report.out_format.code == 'oo-dbf':
+            data = docs.convert(identifier=token)  # TODO What format?
         else:
             data = docs.convert(identifier=token,
                                 out_mime=mime_dict[report.out_format.code],
                                 in_mime=mime_dict[report.in_format]
-                               )
+                                )
         return data
-    
+
     def _get_lang(self, source='current'):
-        if source=='current':
+        if source == 'current':
             return self.env.context['lang'] or self.env.context['user_lang']
-        elif source=='company':
+        elif source == 'company':
             return self.env.user.company_id.partner_id.lang
-        elif source=='user':
+        elif source == 'user':
             return self.env.context['user_lang']
-    
+
     def _set_lang(self, lang, obj=None):
         self.localcontext.update(lang=lang)
         if obj is None and 'objects' in self.localcontext:
             obj = self.localcontext['objects']
         if obj and obj.env.context['lang'] != lang:
             ctx_copy = dict(self.env.context)
             ctx_copy.update(lang=lang)
-            obj.env.context=frozendict(ctx_copy)
+            obj.env.context = frozendict(ctx_copy)
             obj.invalidate_cache()
-    
-    def _format_lang(self, value, digits=None, grouping=True, monetary=False,
-                     dp=False, currency_obj=False):
-        return odoo_fl(self.env, value, digits, grouping, monetary, dp,
-                       currency_obj)
-    
+
+    def _format_lang(
+            self, value, digits=None, grouping=True, monetary=False, dp=False,
+            currency_obj=False, date=False, date_time=False):
+        """ We add date and date_time for backwards compatibility. Odoo has
+        split the method in two (formatlang and format_date)
+        """
+        if date or date_time:
```

Comment:
new odoo formatlang method does not accepts date and datetime, we do this to keep compatibiliy with reports on v8/v9